src/main/java/com/minecolonies/core/entity/ai/workers/util/BuildingStructureHandler.java

```diff
@@ -98,7 +98,16 @@ public BuildingStructureHandler(
         this.workOrder = (IBuilderWorkOrder) workOrder;
         this.structureAI = entityAIStructure;
         this.stages = stages;
-        this.stage = workOrder.getStage();
+
+        for (int i = 0; i < stages.length; i++)
+        {
+            final BuildingProgressStage stage = stages[i];
+            if (stage == workOrder.getStage())
+            {
+                this.stage = i;
```

Comment:
Why do we need to loop match here?

Reply:
Can't we just save the stage here as well than as the enum value?

Reply:
I dont think so since the array determines the stage order, not the enum. But it is your code, ask yourself? :D

Reply:
No, I mean, the loop you do to assign a stage, we could just store the stage in the class instead of the int version of it.